src/Window.scrollBy.ts

```diff
@@ -1,32 +1,30 @@
-import { IAnimationOptions, IScrollToOptions } from "./common.js";
-import { getOriginalFunc, windowScroll } from "./Window.scroll.js";
+import { IAnimationOptions, IScrollToOptions, original, supportsScrollBehavior } from "./common.js";
+import { windowScroll } from "./Window.scroll.js";
 
 export const windowScrollBy = (options: IScrollToOptions) => {
     const left = (options.left || 0) + (window.scrollX || window.pageXOffset);
     const top = (options.top || 0) + (window.scrollY || window.pageYOffset);
 
     if (options.behavior !== "smooth") {
-        return getOriginalFunc()(left, top);
+        return original.windowScroll.call(window, left, top);
     }
 
     return windowScroll({ ...options, left, top });
 };
 
-let $original: (x: number, y: number) => void;
-
 export const windowScrollByPolyfill = (options?: IAnimationOptions) => {
-    const originalFunc = (() => {
-        if ($original === undefined) {
-            $original = window.scrollBy.bind(window);
-        }
-        return $original;
-    })();
+    if (supportsScrollBehavior()) {
+        return;
+    }
 
     window.scrollBy = function scrollBy() {
         const [arg0 = 0, arg1 = 0] = arguments;
 
         if (typeof arg0 === "number" && typeof arg1 === "number") {
```

Comment:
There is uneeded call with default `0` argument, and also `arg0`, `arg1` variables should be more descriptive.

Maybe we can change it, to something like this:

```js
const [xAxis, yAxis] = arguments;
const isValidAxis = axis => typeof axis === "number" && axis > 0;

if (isValidAxis(xAxis) && isValidAxis(yAxis)) {
  return windowScrollBy({ left: xAxis, top: yAxis });
}

Reply:
There is one more thing `window.scrollBy` accepts Numbers and also Number of Strings (like `'123'`). I'd would be great to support both. Maybe it's a good challange to use typescript here, and valid both types?

Reply:
Thanks a lot! You found the bug.
We should to make sure that the polyfill function behaves as same as the standard native function at runtime.